File: backend/src/services/pdf.service.js
import fs from "fs";
import pdfParse from "pdf-parse";

class PdfService {
  async extractText(filePath) {
    const buffer = fs.readFileSync(filePath);

    const data = await pdfParse(buffer);

    return data.text;
  }
}

export default new PdfService();

File: ai-service/app/services/pdf_service.py
from pypdf import PdfReader


class PDFService:

    def extract_text(self, file_path: str):

        reader = PdfReader(file_path)

        text = ""

        for page in reader.pages:
            page_text = page.extract_text()

            if page_text:
                text += page_text + "\n"

        return text


pdf_service = PDFService()